x-pack/plugins/siem/public/components/page/hosts/first_last_seen_host/index.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+import moment from 'moment';
+import React from 'react';
+import { ApolloConsumer } from 'react-apollo';
+import { pure } from 'recompose';
+
+import { useFirstLastSeenHostQuery } from '../../../../containers/hosts/first_last_seen';
+import { getEmptyTagValue } from '../../../empty_value';
+import { PreferenceFormattedDate } from '../../../formatted_date';
+import { LocalizedDateTooltip } from '../../../localized_date_tooltip';
+
+export type FirstLastSeenHostType = 'first-seen' | 'last-seen';
+
+export const FirstLastSeenHost = pure<{ hostname: string; type: FirstLastSeenHostType }>(
+  ({ hostname, type }) => {
+    return (
+      <ApolloConsumer>
+        {client => {
+          const { loading, firstSeen, lastSeen } = useFirstLastSeenHostQuery(
+            hostname,
+            'default',
+            client
+          );
+          return (
+            <>
+              {loading && <EuiLoadingSpinner size="m" />}
+              {!loading &&
+              type === 'first-seen' &&
+              firstSeen != null &&
+              new Date(firstSeen).toString() === 'Invalid Date'
+                ? firstSeen
+                : !loading &&
+                  type === 'first-seen' &&
+                  firstSeen != null && (
+                    <LocalizedDateTooltip date={moment(new Date(firstSeen)).toDate()}>
+                      <PreferenceFormattedDate value={new Date(firstSeen)} />
+                    </LocalizedDateTooltip>
+                  )}
+              {!loading &&
+              type === 'last-seen' &&
+              lastSeen != null &&
+              new Date(lastSeen).toString() === 'Invalid Date'
+                ? lastSeen
+                : !loading &&
+                  type === 'last-seen' &&
+                  lastSeen != null && (
+                    <LocalizedDateTooltip date={moment(new Date(lastSeen)).toDate()}>
+                      <PreferenceFormattedDate value={new Date(lastSeen)} />
+                    </LocalizedDateTooltip>
+                  )}
+              {!loading && firstSeen == null && getEmptyTagValue()}
```

Comment:
You are not checking for `type` here so this can output something like this with double dashes on the end which I don't think you want.

```ts
<span class="euiToolTipAnchor">Apr 8, 2019 @ 18:35:45.064</span>--
```

I think you want this logic:
```ts
{!loading && type === 'first-seen' && firstSeen == null && getEmptyTagValue()}
{!loading && type === 'last-seen' && lastSeen == null && getEmptyTagValue()}
```

Two unit tests for each condition which currently fail but are fixed with the above code:

```ts
  test('First Seen is empty but not Last Seen', async () => {
    const badDateTime = cloneDeep(mockFirstLastSeenHostQuery);
    // tslint:disable-next-line:no-any
    (badDateTime[0] as any).result.data.source.HostFirstLastSeen.firstSeen = null;
    const { container } = render(
      <TestProviders>
        <MockedProvider mocks={badDateTime} addTypename={false}>
          <FirstLastSeenHost hostname="kibana-siem" type="last-seen" />
        </MockedProvider>
      </TestProviders>
    );

    await wait();

    expect(container.innerHTML).toBe(
      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 18:35:45.064</span>'
    );
  });

  test('Last Seen is empty but not First Seen', async () => {
    const badDateTime = cloneDeep(mockFirstLastSeenHostQuery);
    // tslint:disable-next-line:no-any
    (badDateTime[0] as any).result.data.source.HostFirstLastSeen.lastSeen = null;
    const { container } = render(
      <TestProviders>
        <MockedProvider mocks={badDateTime} addTypename={false}>
          <FirstLastSeenHost hostname="kibana-siem" type="first-seen" />
        </MockedProvider>
      </TestProviders>
    );

    await wait();

    expect(container.innerHTML).toBe(
      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 16:09:40.692</span>'
    );
  });
```